### ISSUE TITLE
Project Docs: Define Branching Strategy

### ISSUE THREAD

**Issue by Jose-Servin** [Jose-Servin/NLB#9]
Define the branching strategy in the project README

For effective development and integration, we use a structured branching strategy. Here’s a quick guide:

- **`main`**: The production-ready branch that contains stable code. All final releases are merged into this branch.
  
- **`staging`**: The branch used for pre-production testing. Features that are integrated and tested in `dev` are merged into `staging` for final validation before being released to production.

- **`dev`**: The branch where active development occurs. Feature branches are merged into `dev` for integration and initial testing. Once features are stable, they are merged into `staging`.

- **`feature/[feature-name]`**: Individual branches for developing new features or fixes. These branches are created off of `dev` and are merged back into `dev` once the feature is complete and tested.

### Workflow

1. **Create Feature Branches**: Start by creating a feature branch off of `dev` for each new feature or fix. Use the naming convention `feature/[feature-name]`.

2. **Develop and Test**: Work on the feature in its own branch. Once complete, merge it back into `dev` for integration and further testing.

3. **Prepare for Production**: Once `dev` is stable, merge it into `staging` for final pre-production testing.

4. **Deploy to Production**: After successful testing in `staging`, merge `staging` into `main` for production release.
